### ISSUE TITLE
fb is duplicating events and purchases ....all purchases are getting duplicated 

### ISSUE THREAD

**Issue by rizwan1209** [woocommerce/facebook-for-woocommerce#2474]
## 🔎 Isolate the bug

- [x] I have confirmed this occurs in the most recent version of WordPress, WooCommerce, and Facebook for WooCommerce.
- [x] I have confirmed this occurs when only WooCommerce and Facebook for WooCommerce are active and when using a default WordPress or WooCommerce theme.

## 🚶‍♀️ Steps to reproduce

1. The first step
1. The next step… 

## 🗃 Logs

<details>
	
</details>

**Comment by rawdreeg** [comment]
Hi @rizwan1209,

Could you give us more details on this issue? Feel free to add appropriate screenshots that display the bug. 

It would also be helpful if you included logs, details about your WordPress environment (from [**WooCommerce Status Report**](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/)), and any other relevant information about your site.

**Comment by rizwan1209** [comment]
`
### WordPress Environment ###

WordPress address (URL): https://www.popstripesonline.com
Site address (URL): https://www.popstripesonline.com
WC Version: 7.3.0
REST API Version: ✔ 7.3.0
WC Blocks Version: ✔ 9.1.5
Action Scheduler Version: ✔ 3.5.2
Log Directory Writable: ✔
WP Version: 6.1.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 8.0.27
PHP Post Max Size: 512 MB
PHP Time Limit: 60
PHP Max Input Vars: 10000
cURL Version: 7.84.0
OpenSSL/1.1.1p

SUHOSIN Installed: –
MySQL Version: 10.3.38-MariaDB
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.3.0
WC Database Prefix: fwp_
Total Database Size: 178.93MB
Database Data Size: 87.41MB
Database Index Size: 91.52MB
fwp_woocommerce_sessions: Data: 1.48MB + Index: 0.02MB + Engine InnoDB
fwp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fwp_woocommerce_order_items: Data: 0.42MB + Index: 0.13MB + Engine InnoDB
fwp_woocommerce_order_itemmeta: Data: 3.52MB + Index: 5.03MB + Engine InnoDB
fwp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fwp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_actionscheduler_actions: Data: 3.02MB + Index: 7.36MB + Engine InnoDB
fwp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_actionscheduler_logs: Data: 2.02MB + Index: 2.03MB + Engine InnoDB
fwp_adtribes_my_conversions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_aepc_custom_audiences: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_aepc_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_atum_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_atum_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_atum_product_data: Data: 1.52MB + Index: 0.36MB + Engine InnoDB
fwp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_comments: Data: 1.52MB + Index: 0.95MB + Engine InnoDB
fwp_easypay_order: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_jazz_cash_order_ref: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_mailchimp_carts: Data: 0.33MB + Index: 0.00MB + Engine InnoDB
fwp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_nx_entries: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
fwp_nx_posts: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fwp_nx_stats: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
fwp_options: Data: 7.45MB + Index: 2.06MB + Engine InnoDB
fwp_postmeta: Data: 47.23MB + Index: 65.44MB + Engine InnoDB
fwp_posts: Data: 8.42MB + Index: 1.97MB + Engine InnoDB
fwp_sbi_feeds: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_sbi_feed_caches: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_sbi_instagram_feeds_posts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_sbi_instagram_feed_locator: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_sbi_instagram_posts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_sbi_sources: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_social_users: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_stock_log: Data: 0.52MB + Index: 0.00MB + Engine InnoDB
fwp_termmeta: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
fwp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_term_relationships: Data: 1.20MB + Index: 1.25MB + Engine InnoDB
fwp_term_taxonomy: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
fwp_usermeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
fwp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fwp_viredis_product_pricing: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_vi_wad_ali_orders_info: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_vi_wad_error_product_images: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wccs_conditions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wccs_condition_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_wcpdf_invoice_number: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
fwp_wcpdf_packing_slip_number: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
fwp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wc_customer_lookup: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
fwp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_wc_order_product_lookup: Data: 0.44MB + Index: 0.36MB + Engine InnoDB
fwp_wc_order_stats: Data: 0.16MB + Index: 0.16MB + Engine InnoDB
fwp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_wc_product_attributes_lookup: Data: 0.14MB + Index: 0.09MB + Engine InnoDB
fwp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_wc_product_meta_lookup: Data: 1.52MB + Index: 1.00MB + Engine InnoDB
fwp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_wdp_orders: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fwp_wdp_order_items: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
fwp_wdp_persistent_rules_cache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wdp_rules: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_wdr_order_discounts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wdr_order_item_discounts: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
fwp_wdr_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wt_iew_action_history: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
fwp_wt_iew_cron: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wt_iew_ftp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_wt_iew_mapping_template: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
fwp_yith_wcwl: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_yith_wcwl_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
fwp_yoast_indexable: Data: 4.28MB + Index: 1.44MB + Engine InnoDB
fwp_yoast_indexable_hierarchy: Data: 0.11MB + Index: 0.14MB + Engine InnoDB
fwp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fwp_yoast_primary_term: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
fwp_yoast_seo_links: Data: 0.17MB + Index: 0.09MB + Engine InnoDB

### Post Type Counts ###

attachment: 988
blocks: 3
custom_css: 1
elementor_library: 1
nav_menu_item: 55
notificationx: 1
oembed_cache: 2
page: 12
post: 18
product: 442
product_variation: 2338
revision: 355
shop_coupon: 12
shop_order: 1360
shop_order_refund: 116
size-chart: 229
wp_global_styles: 1
wpcf7_contact_form: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (29) ###

Easypay: by Systems Limited – 0.1
ATUM Inventory Management for WooCommerce: by Stock Management Labs™ – 1.9.26.2
Code Snippets: by Code Snippets Pro – 3.2.2
CommentLuv: by Elegant Digital Solutions – 3.0.4
Contact Form 7: by Takayuki Miyoshi – 5.7.3
Customer Reviews for WooCommerce: by CusRev – 5.18.0
GTM4WP: by Thomas Geiger – 1.16.2
Facebook for WooCommerce: by Facebook – 3.0.10
Site Kit by Google: by Google – 1.93.0
Smash Balloon Instagram Feed: by Smash Balloon – 6.1.1
Jetpack: by Automattic – 11.8
Mailchimp for WooCommerce: by Mailchimp – 2.7.6
NotificationX: by WPDeveloper – 2.6.0
reSmush.it Image Optimizer: by reSmush.it – 0.4.11
SiteGround Optimizer: by SiteGround – 7.3.0
Advanced Editor Tools (previously TinyMCE Advanced): by Automattic – 5.6.0
Import Export WordPress Users and WooCommerce Customers: by WebToffee – 2.3.7
CTX Feed: by WebAppick – 5.0.0
Product Size Charts Plugin for WooCommerce Premium: by theDotstore – 2.4.1
Checkout Field Editor for WooCommerce: by ThemeHigh – 1.8.1
Variation Swatches for WooCommerce: by Emran Ahmed – 2.0.18
WooCommerce Google Analytics Integration: by WooCommerce – 1.6.0
PDF Invoices & Packing Slips for WooCommerce: by WP Overnight – 3.3.2
WooCommerce Shipping & Tax: by WooCommerce – 2.2.2
Stock Manager for WooCommerce: by StoreApps – 2.10.0
WooCommerce: by Automattic – 7.3.0
Yoast SEO: by Team Yoast – 20.1
SiteGround Central: by SiteGround – 2.0.9
YITH WooCommerce Wishlist: by YITH – 3.17.0

### Inactive Plugins (0) ###


### Settings ###

API Enabled: –
Force SSL: –
Currency: PKR (₨)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
Order datastore: WC_Order_Data_Store_CPT

### WC Pages ###

Shop base: #46 - /shop/
Cart: #47 - /cart/
Checkout: #48 - /checkout/
My account: #49 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Flatsome
Version: 3.16.4
Author URL: https://uxthemes.com
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php version 3.6.0 is out of date. The core version is 7.3.0
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php
flatsome/woocommerce/content-product_cat.php

Outdated Templates: ❌
					
					
						Learn how to update

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2023-02-13 18:28:59 +00:00
Options: ✔
Notes: 56
Onboarding: completed

### Action Scheduler ###

Canceled: 3
Oldest: 2023-01-27 09:24:58 +0000
Newest: 2023-01-27 10:25:57 +0000

Complete: 5,648
Oldest: 2023-01-13 08:48:36 +0000
Newest: 2023-02-13 08:05:49 +0000

Failed: 14
Oldest: 2021-11-23 13:49:39 +0000
Newest: 2023-01-14 07:10:34 +0000

Pending: 4
Oldest: 2023-02-13 09:58:24 +0000
Newest: 2023-02-14 09:08:05 +0000


### Status report information ###

Generated at: 2023-02-13 08:40:32 +00:00
`

**Comment by rawdreeg** [comment]
Thanks, @rizwan1209, for providing additional details. We'll be investigating this issue.

**Comment by layoutd** [comment]
Hi @rizwan1209, if you view the source code of the you checkout page (https://www.popstripesonline.com/checkout/)>, you'll see that the `fbq('track', 'InitiateCheckout'` event is triggered two times. The first time indicates `<!-- Facebook Pixel Code -->` around line 200, and the second time indicates `<!-- WooCommerce JavaScript -->` and `/* WooCommerce Facebook Integration Event Tracking */` around line 1200. 

This second snippet is the one generated by the Facebook for WooCommerce extension, but the first snippet looks the code generated by Facebook when "Manually Install Pixel Code Yourself" is selected during Pixel installation ([example installation instructions](https://adwisely.com/blog/how-to-install-pixel-on-woocommerce-store/#Manually_install)). It seems likely that this tracking code being added by another extension or manually. Perhaps the **Code Snippets Pro** extension is being used to manually add Facebook event triggers, or the theme templates have been modified to include this code on the checkout page?

**Comment by rizwan1209** [comment]
Dear concern

How do i fix this problem and how do i remove manually installed one kindly guide step by step since im not a developer and will need your guidance 

Regards

**Comment by layoutd** [comment]
Hi @rizwan1209, to seek further help with this issue,  please reach out to WooCommerce support. The best place to do so is the [WordPress.org Facebook for WooCommerce forum](https://wordpress.org/support/plugin/facebook-for-woocommerce/). Or, if you have a WooCommerce.com account, you can [start a chat or open a ticket on WooCommerce.com](https://woocommerce.com/my-account/create-a-ticket/). I'll close this issue as you seek support in one of those channels.

**Comment by rizwan1209** [comment]
I HAVE ANOTHER ISSUE WHICH IS:

MY PRODUCTS ARE NOT SYNCING ON FACEBOOK SHOP AS WHEN I CHANGED PRICE AND PICTURE OF A PRODUCT ON WEBSITE BUT IT DID NOT SYNC TO FACEBOOK PAGE/SHOP AS IT STILL SHOWS OLD PRICE AND OLD PICTURE..

I ALSO PRESSED PRODUCT SYNC IN FACEBOOK FOR WOOCOMERCE PLUGIN BUT THAT ALSO DID NOT HELP

**Comment by atty31** [comment]
I have a similar issue. The Purchase event is fired multiple times on the thank you page. 
Also it is fired if i go  from the cart page to the checkout page .Or if I submit the checkout form, however i have validation errors from the shipping address, the purchase event is still fired.  Could this help ?
